src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1479,16 +1551,38 @@ void ICanSaveModel.Save(ModelSaveContext ctx)
             for (int index = 0; index < memberInfos.Length; index++)
             {
                 var memberInfo = memberInfos[index];
-                var mappingAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
 
-                if (mappingAttr == null)
-                    throw host.Except($"{(memberInfo is FieldInfo ? "Field" : "Property")} '{memberInfo.Name}' is missing the {nameof(LoadColumnAttribute)} attribute");
+                var ignoreLoadColumnAttr = memberInfo.GetCustomAttribute<LoadColumnIgnoreAttribute>();
+                if (ignoreLoadColumnAttr != null && ignoreLoadColumnAttr.Ignore)
+                    continue;
+
+                var mappingLoadColumnAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
+                var mappingLoadColumnNameAttr = memberInfo.GetCustomAttribute<LoadColumnNameAttribute>();
+                var memberInfoType = memberInfo is FieldInfo ? "Field" : "Property";
+
+                if (!hasHeader && mappingLoadColumnAttr == null)
+                    throw host.Except(
+                        $"You should set {nameof(hasHeader)} to true in order to use {nameof(LoadColumnNameAttribute)}. " +
+                        $"To automatically map the name of {memberInfoType} to {nameof(LoadColumnNameAttribute)}, set {nameof(autoMapLoadColumns)} to true. " +
+                        $"To specify a fixed column index, use {nameof(LoadColumnAttribute)}. " +
```

Comment:
If autoMalLoadColumns is true, then the hasHeader must be true. 
Can we factor this out of the loop as an explicit parameter check at the beginning of the function?

Reply:
I think that other than to be able to factor this check out of the loop, the `autoMapLoadColumns` argument is not needed, so I would prefer not to have it in the signature of the method at all.

---
In reply to: [409848508](https://github.com/dotnet/machinelearning/pull/5021#discussion_r409848508) [](ancestors = 409848508)